src/main/kotlin/com/noumenadigital/npl/cli/CommandProcessor.kt

```diff
@@ -1,34 +1,37 @@
 package com.noumenadigital.npl.cli
 
+import com.noumenadigital.npl.cli.commands.registry.BaseDirectoryAware
+import com.noumenadigital.npl.cli.exception.CommandExecutionException
 import com.noumenadigital.npl.cli.exception.CommandNotFoundException
 import com.noumenadigital.npl.cli.exception.CommandParsingException
 import com.noumenadigital.npl.cli.exception.InternalException
 import com.noumenadigital.npl.cli.exception.buildOutputMessage
 import com.noumenadigital.npl.cli.service.CommandsParser
 import java.io.Writer
+import java.nio.file.Path
 
 class CommandProcessor(
     private val commandsParser: CommandsParser = CommandsParser,
+    private val baseDir: Path? = null,
 ) {
-    companion object {
-        private const val START_MESSAGE = "Executing command '%s'...\n"
-        private const val END_MESSAGE_SUCCESS = "\nCommand '%s' finished SUCCESSFULLY."
-    }
-
     fun process(
         inputArgs: List<String>,
         output: Writer,
     ) {
         output.use { out ->
             try {
                 val command = commandsParser.parse(inputArgs)
-                out.write(START_MESSAGE.format(command.commandName))
+
+                if (baseDir != null && command is BaseDirectoryAware) {
+                    command.setBaseDirectory(baseDir)
+                }
+
                 command.execute(out)
-                out.write(END_MESSAGE_SUCCESS.format(command.commandName))
             } catch (ex: InternalException) {
                 when (ex) {
                     is CommandNotFoundException -> output.write(ex.buildOutputMessage())
                     is CommandParsingException -> output.write(ex.buildOutputMessage())
+                    is CommandExecutionException -> output.write(ex.buildOutputMessage(inputArgs))
```

Comment:
More on the entire function, but shouldn't we somewhere deal with exit codes? 

Now we basically support "failed successfully"

Reply:
Good point, I thought about it and then forgot about it 😇

Reply:
![image](https://github.com/user-attachments/assets/939ef774-4981-47ef-9e79-108e249f5577)